### PR TITLE
chore(test): migrate from jest to vitest

### PR DESCRIPTION
**Pull request by philibea**
## Type

- Migration

#### What is expected?

From jest to vitest

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -185,6 +189,12 @@
     "timekeeper": "2.3.1",
     "typescript": "5.4.5",
     "vite": "5.2.11",
+    "vitest": "1.6.0",
+    "vitest-canvas-mock": "0.3.3",
+    "vitest-localstorage-mock": "0.1.2",
     "zxcvbn": "4.4.2"
+  },
+  "dependencies": {
+    "happy-dom": "^14.7.1"
```
Comment (matthprost) [inline package.json:198]:
```suggestion
    "happy-dom": "14.7.1"
```

**Review by matthprost — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -185,6 +173,12 @@
     "timekeeper": "2.3.1",
     "typescript": "5.4.5",
     "vite": "5.2.11",
+    "vitest": "1.6.0",
+    "vitest-canvas-mock": "0.3.3",
+    "vitest-localstorage-mock": "0.1.2",
     "zxcvbn": "4.4.2"
+  },
+  "dependencies": {
+    "vitest-axe": "^0.1.0"
```
Comment (matthprost) [inline package.json:182]:
```suggestion
    "vitest-axe": "0.1.0"
```

File: packages/illustrations/vitest.setup.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import '@testing-library/jest-dom/vitest'
+import { cleanup } from '@testing-library/react'
+import { setup } from '@utils/test'
+import { afterAll, afterEach, beforeAll, expect, vi } from 'vitest'
+
+expect.extend(matchers)
+
+setup()
+
+beforeAll(() => {
```
Comment (matthprost) [inline packages/illustrations/vitest.setup.ts:13]: Can't we move this file into `@utils/test`?

**Review by matthprost — CHANGES_REQUESTED** [review]

File: packages/illustrations/vitest.setup.ts (hunk 3 above)
Comment (philibea) [reply packages/illustrations/vitest.setup.ts]: Yes done !

**Review by johnrazeur — APPROVED** [review]

**Pull request merged**